source/server/listener_manager_impl.cc

```diff
@@ -379,11 +374,33 @@ std::vector<std::reference_wrapper<Listener>> ListenerManagerImpl::listeners() {
   return ret;
 }
 
+void ListenerManagerImpl::addListenerToWorker(Worker& worker, ListenerImpl& listener) {
+  worker.addListener(listener, [this, &listener](bool success) -> void {
+    // The add listener completion runs on the worker thread. Post back to the main thread to
+    // avoid locking.
+    server_.dispatcher().post([this, success, &listener]() -> void {
+      // It is theoretically possible for a listener to get added on 1 worker but not the others.
+      // The following code will yield 1 critical log and 1 stat. It will also cause listener
+      // removal a single time. Note also that that drain/removal can race with addition. It's
+      // guaranteed that workers process remove after add so this should be fine.
+      if (!success && !listener.onListenerCreateFailure()) {
+        // TODO(mattklein123): In addition to a critical log and a stat, we should consider adding
+        //                     a startup option here to cause the server to exit. I think we
+        //                     probably want this at Lyft but I will do it in a follow up.
```

Comment:
I don't think we would actually add a listener all the way to this point in a dry run situation? It's not completely clear to me what happens in a dry run (I saw your other comment). I think we should track this separately?